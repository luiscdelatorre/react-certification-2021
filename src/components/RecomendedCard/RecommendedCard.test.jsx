import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RecommendedCard from './RecomendedCard.component';
import relatedVideos from '../../mock/youtube-related-videos-mock.json';

describe('RecommendedCard', () => {
  it('Should display mock data in component', () => {
    const video = relatedVideos.items[0];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const id = video.id.videoId;
    render(
      <BrowserRouter>
        <RecommendedCard
          id={id}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );

    expect(screen.getByText(video.snippet.title).tagName).toBe('H5');
    expect(screen.getByText(video.snippet.channelTitle).tagName).toBe('H6');
  });
  it('Should render conditional item for liveBroadcastContent', () => {
    const video = relatedVideos.items[4];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const id = video.id.videoId;
    render(
      <BrowserRouter>
        <RecommendedCard
          id={id}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );

    expect(screen.getByText(video.snippet.title).tagName).toBe('H5');
    expect(screen.getByText(video.snippet.channelTitle).tagName).toBe('H6');
    expect(screen.getByText('live')).toBeInTheDocument();
  });

  it('Validates VideoCard snapshot', () => {
    const video = relatedVideos.items[0];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const id = video.id.videoId;
    const component = renderer.create(
      <BrowserRouter>
        <RecommendedCard
          id={id}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
