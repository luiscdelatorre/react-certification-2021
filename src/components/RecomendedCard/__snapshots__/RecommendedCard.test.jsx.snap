// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendedCard Validates VideoCard snapshot 1`] = `
<a
  className="sc-bdfBwQ iEkQBl"
  href="/video/wodwkc9bzqY"
  onClick={[Function]}
>
  <div
    className="sc-gsTCUz gbEMKB"
  >
    <img
      alt="SNAKES AND LADDERS | Comedy Family Challenge | Biggest Saap Sidi | Aayu and Pihu Show"
    />
    
  </div>
  <div
    className="sc-dlfnbm fOolBg"
  >
    <h5
      className="sc-eCssSg eMkQAo"
      title="SNAKES AND LADDERS | Comedy Family Challenge | Biggest Saap Sidi | Aayu and Pihu Show"
    >
      SNAKES AND LADDERS | Comedy Family Challenge | Biggest Saap Sidi | Aayu and Pihu Show
    </h5>
    <h6
      className="sc-jSgupP hHNFVW"
    >
      Aayu and Pihu Show
    </h6>
  </div>
</a>
`;
